dask/dataframe/core.py

```diff
@@ -1177,6 +1177,62 @@ def repartition(
         elif freq is not None:
             return repartition_freq(self, freq=freq)
 
+    def shuffle(
+        self,
+        on,
+        npartitions=None,
+        max_branch=None,
+        shuffle=None,
+        ignore_index=False,
+        compute=None,
+    ):
+        """ Rearrange DataFrame into new partitions
+
+        Uses hashing of `on` to map rows to output partitions. After this
+        operation, rows with the same index element(s) will be in the same
+        partition.
+
+        Parameters
+        ----------
+        on : str, list of str, or Series, Index, or DataFrame
+            Column(s) or index to be used to map rows to output partitions
+        npartitions : int, optional
+            Number of partitions of output. Partition count will not be
+            changed by default.
+        max_branch: int, optional
+            The maximum number of splits per input partition. Used within
+            the staged shuffling algorithm.
+        shuffle: {'disk', 'tasks'}, optional
+            Either ``'disk'`` for single-node operation or ``'tasks'`` for
+            distributed operation.  Will be inferred by your current scheduler.
+        ignore_index: bool, default False
+            Ignore index during shuffle.  If ``True``, performance may improve,
+            but index values will not be preserved.
+        compute: bool
+            Whether or not to trigger an immediate computation. Defaults to False.
+            Note, that even if you set ``compute=False``, an immediate computation
+            will still be triggered if ``divisions`` is ``None``.
```

Comment:
Does `divisions` here refert to `self.divisions`? Or is this supposed to refer to a parameter?

Reply:
Ah - Sorry, this was a copy-paste mistake from the set_index description of `compute` - Since we are using hash values (rather than sorted order), there is no need to pass in divisions.